gridmet_split_script.py

```diff
@@ -0,0 +1,219 @@
+                           lon = slice(lon_min, lon_max),
+                           lat = slice(lat_max, lat_min))
+        end = time.perf_counter()
+        print(f'finish in {round(start - end, 2)} seconds')
+
+        # Append to dict of xr.datasets
+        xarray_dict[var] = ds_subset
+
+    return xarray_dict
+
+
+def create_weightmap(xarray_dict, polygon, output_data_folder, weightmap_var = None):
+
+    """
+    :param geodataframe polygon: geodataframe polygon or multipolygon
+    :param dict xarray_dict: dictionary of gridmet data. the output of get_gridmet_datasets()
+    :param str output_data_folder: path to folder for function output
+    :param str weightmap_var: variable used to make weightfile + naming of output e.i. weightmap_var = 'tmin'. If none, first variable of dict keys will be used.
+    :return: output path to weightmap pickle file
+    """
+    if isinstance(polygon, geopandas.geodataframe.GeoDataFrame):
+        print('polygon is geodataframe')
+        pass
+    elif os.path.isfile(polygon):
+        print('polygon is path to shapefile')
+        polygon = gpd.read_file(polygon_as_bbox)
+    else:
+        raise TypeError('polygon should str path to shapefile or geodataframe')
+
+    ## Name output weightmap_file
+    if weightmap_var is None:
+        # If non var given
+        weightmap_file = os.path.join(output_data_folder, 'grd2shp_weights.pickle')
+        weightmap_var = list(xarray_dict.keys())[0]
+        print(weightmap_file, weightmap_var)
+    else:
+        weightmap_file = os.path.join(output_data_folder, f'grd2shp_weights_{weightmap_var}.pickle')
+        print(weightmap_file)
+
+    ## Produce weightmap
+    start = time.perf_counter()
+    weightmap = xa.pixel_overlaps(xarray_dict[weightmap_var], polygon)
+    end = time.perf_counter()
+    print(f'finished agg in {round(end - start, 2)} second(s)')
+
+    with open(weightmap_file, 'wb') as file:
+        pickle.dump(weightmap, file)
+
+    return weightmap_file
+
+
+def g2shp_regridding(xarray_dict, weightmap_file, output_data_folder, g2s_file_prefix, g2s_time_var = 'day', g2s_lat_var = 'lat' , g2s_lon_var = 'lon'):
+
+    """
+    :param dict xarray_dict: dictionary of gridmet data. the output of get_gridmet_datasets()
+    :paran str weightmap_file: path to weight file for redridding
+    :param str output_data_folder: path to folder for function output
+    :param str weightmap_var: if len(xarray_dict) selected
+    :param str g2s_time_var: time variable for g2shp initialization (i.e.g2s_time_var = 'day')
+    :param str g2s_lat_var: latitude variable for g2shp initialization (i.e.g2s_time_var = 'lat')
+    :param str g2s_lon_var: time variable for g2shp initialization (i.e.g2s_time_var = 'lon')
+    :return:
+    """
+
+    ## params for g2s.initialise()
+    # grab long name of data vars from dict and place in list
+    vars_long_list = []
+    for key in xarray_dict:
+        vars_long_list = vars_long_list + list(xarray_dict[key])
```

Comment:
I suggest making this a list comprehension for simplicity. I also think it'd be good to put a note about how this works:
```suggestion
# list(xarray.Dataset) outputs a list of variable names. We only take the first because 
# the gridmet datasets only have one variable (e.g., "daily_maximum_temperature")
vars_long_list = [list(xarray_dict[key])[0] for key in xarray_dict]
```

Reply:
Ah nice, definitely agree list comp is cleaner. Not always my go-to format for loops. How come the `[0]` works to keep in a single list?